plonky2x/src/frontend/builder/mux.rs

```diff
@@ -0,0 +1,71 @@
+use plonky2::field::extension::Extendable;
+use plonky2::hash::hash_types::RichField;
+
+use crate::prelude::{CircuitBuilder, CircuitVariable, Variable};
+
+trait MuxBuilder {
+    fn select_index<T: CircuitVariable>(&mut self, selector: Variable, inputs: &[T]) -> T;
+}
+
+impl<F: RichField + Extendable<D>, const D: usize> MuxBuilder for CircuitBuilder<F, D> {
```

Comment:
nit: here I think you can just remove the trait `MuxBuilder` and implement it directly on `CircuitBuilder`--our mistake for including that in the original template

Reply:
yeah, that's also what feel a bit weird for me : )